language/stdlib/modules/DiemAccount.move

```diff
@@ -1489,6 +1492,31 @@ module DiemAccount {
         aborts_if !DiemTransactionPublishingOption::spec_is_module_allowed(sender) with Errors::INVALID_STATE;
     }
 
+    /// If published, halts transactions from all accounts except DiemRoot
+    resource struct HaltAllTransactions {}
```

Comment:
Not related to this PR, but this `HaltAllTransactions` might also solve the problem of publishing/republishing a module invalidates code cache (#7367) if we issue one in the `module_epilogue`?

Reply:
I might be misunderstanding the use-case, but I think we would probably not want to do this. Because this stops all payments from going through, it's an extreme measure that should only be used in emergencies.